File: 백준/단계 별로 풀어보기/1차원 배열/1546_평균.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int main()
{
	int N;
	scanf("%d", &N);
	int *score = new int[N];
	
	int sum = 0;
	int max = 0;
	for (int i = 0; i < N; i ++)
	{
	    scanf("%d", &score[i]);
	    if (max < score[i]) max = score[i];
	    sum += score[i];
	}
	printf("%lf", (float)(sum) / (max * N) * 100);
}

File: baekjoon/단계 별로 풀어보기/1차원 배열/1546_평균.py
import sys
input = sys.stdin.readline
IP = lambda: map(int, sys.stdin.readline().split())

def solve():
    N = int(input())
    score = list(IP())
    print(sum(score) / max(score) * 100 / N)
    
if "__main__" == __name__: solve()